### ISSUE TITLE
bug of executing index scan

### ISSUE THREAD

**Issue by mengranwo** [cmu-db/peloton#600]
We found a bug on optimizer when executing index scan.  If there exists several different indexes on the same table, the optimizer may fail to choose the right index, thus return incomplete or incorrect results. 

postgres=# create table foo(a int primary key, b varchar(32), c int);
create table foo(a int primary key, b varchar(32), c int) 0

postgres=# create unique index sk0 on foo(b,c);
create unique index sk0 on foo(b,c) 0

postgres=# create index sk1 on foo(c);
create index sk1 on foo(c) 0

postgres=# select * from foo;
 a  |   b   | c  
----+-------+----
  1 | look  |  2
  **3 | looks |  4
  8 | good  |  4**
  2 | good  |  5
 -1 | good  | -5
(5 rows)

postgres=# select * from foo where c = 4; 
 a |  b   | c 
---+------+---
 8 | good | 4
(1 row)
**The correct answer should be 2 rows instead of 1.**

**Comment by apavlo** [comment]
It looks like we are creating a UNIQUE index on 'c' even though you don't ask for it:

```
postgres=# create index sk1 on foo(c);
create unique index sk1 on foo(c) 0
```

**Comment by AngLi-Leon** [comment]
Another one:
```
postgres=# create table foo(a int primary key, b varchar(32) primary key);
create table foo(a int primary key, b varchar(32) primary key) 
postgres=# create index sk0 on foo(a);
create index sk0 on foo(a) 0
postgres=# insert into foo values(2, '323');
insert into foo values(2, '323') 1
postgres=# insert into foo values(2, '313');
insert into foo values(2, '313') 1
postgres=# insert into foo values(1, '313');
insert into foo values(1, '313') 1
postgres=# select * from foo;
 a |  b  
---+-----
 2 | 323
 2 | 313
 1 | 313
(3 rows)

postgres=# select * from foo where b = '313';
 a |  b  
---+-----
 1 | 313
 2 | 313
 2 | 323
(3 rows)
```

**Comment by mengranwo** [comment]
Oh, sorry that was an typo. It was a non unique index. 
The default value for index is non unique if don't using the key word "unique".

**Comment by AngLi-Leon** [comment]
Looks like when we have an index on 2 attributes and select based on one of them, the optimizer selects the wrong index...

**Comment by apavlo** [comment]
It also looks like we still have the wrong PRIMARY KEY syntax...

**Comment by AngLi-Leon** [comment]
Yeah.... You get me....

**Comment by AngLi-Leon** [comment]
@https://github.com/cmu-db/peloton/pull/592/commits/92f3e09cf8a2ed5750b462191d8d35c98ee83581

**Comment by AngLi-Leon** [comment]
This is where potentially causes the bug: https://github.com/cmu-db/peloton/pull/592/commits/92f3e09cf8a2ed5750b462191d8d35c98ee83581

**Comment by apavlo** [comment]
Fixed in #716
